code/common/src/test/scala/com/avast/cloud/datadog4s/helpers/RepeatedTest.scala

```diff
@@ -12,23 +13,37 @@ import scala.concurrent.ExecutionContext
 import scala.concurrent.duration._
 
 class RepeatedTest extends AnyFlatSpec with Matchers {
-  private val ec: ExecutionContext                  = scala.concurrent.ExecutionContext.Implicits.global
+  private val ec: ExecutionContext                  = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(5))
   implicit val contextShift: ContextShift[IO]       = cats.effect.IO.contextShift(ec)
   implicit val timer: Timer[IO]                     = IO.timer(ec)
   private val logger                                = LoggerFactory.getLogger(classOf[RepeatedTest])
   private val noopErrHandler: Throwable => IO[Unit] = (_: Throwable) => IO.unit
 
   "repeated test" should "be called repeatedly" in {
+    val waitFor = 10
+
     val test = Ref.of[IO, Int](0).flatMap { ref =>
-      val forever =
+      val endlessProcess =
```

Comment:
I think way cleaner approach is to use `Deffered` and complete it once desired count is reached.  Then you just need to `get` on it.

Reply:
can you share more details? Im not sure how to achieve this without the code being almost the same

Reply:
i tried something in the next commit ... i cold sure use countdown latch in this case